src/draw_target.rs

```diff
@@ -398,13 +398,13 @@ impl std::ops::DerefMut for DrawStateWrapper<'_> {
 impl Drop for DrawStateWrapper<'_> {
     fn drop(&mut self) {
         if let Some(orphaned) = &mut self.orphan_lines {
-            orphaned.extend(
-                self.state
-                    .lines
-                    .iter()
-                    .filter(|l| matches!(l, LineType::Text(_) | LineType::Empty))
-                    .cloned(),
-            );
+            let (text, bars): (Vec<_>, Vec<_>) = self
+                .state
+                .lines
+                .drain(..)
+                .partition(|l| matches!(l, LineType::Text(_) | LineType::Empty));
+            self.state.lines = bars;
+            orphaned.extend(text);
```

Comment:
I'd like to avoid the intermediate allocations from `drain().partition()`. Can we just write this as a `for`-loop that directly pushes into `self.state.lines` and `orphaned` depending on the condition?

Reply:
@djc ping on this one

Reply:
I guess this is a case where'd want `Vec::extract_if()`. Given that that's still unstable, I still think we can avoid allocating the intermediate `Vec` for `orphaned` since we can `push()` onto that directly while draining `lines`, right?